Extension/src/LanguageServer/cppbuildTaskProvider.ts

```diff
@@ -0,0 +1,320 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All Rights Reserved.
+ * See 'LICENSE' in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+import * as path from 'path';
+import * as vscode from 'vscode';
+import * as os from 'os';
+import * as util from '../common';
+import * as telemetry from '../telemetry';
+import { Client } from './client';
+import * as configs from './configurations';
+import * as ext from './extension';
+import * as fs from 'fs';
+import * as nls from 'vscode-nls';
+import * as cp from "child_process";
+
+const localize: nls.LocalizeFunc = nls.loadMessageBundle();
+export const failedToParseTasksJson: string = localize("failed.to.parse.tasks", "Failed to parse tasks.json, possibly due to comments or trailing commas.");
+
+export interface CppBuildTaskDefinition extends vscode.TaskDefinition {
+    type: string;
+    label: string;
+    command: string;
+    args: string[];
+    options: cp.ProcessEnvOptions | undefined;
+}
+
+export class CppBuildTaskProvider implements vscode.TaskProvider {
+    static CppBuildScriptType: string = 'cppbuild';
+    static CppBuildSourceStr: string = "C/C++";
+    private tasks: vscode.Task[] | undefined;
+
+    constructor() {}
+
+    public async provideTasks(): Promise<vscode.Task[]> {
+        if (this.tasks !== undefined) {
+            return this.tasks;
+        }
+        return this.getTasks(false, false);
+    }
+
+    // Resolves a task that has no [`execution`](#Task.execution) set.
+    public resolveTask(_task: vscode.Task): vscode.Task | undefined {
+        const execution: vscode.ProcessExecution | vscode.ShellExecution | vscode.CustomExecution | undefined = _task.execution;
+        if (execution === undefined) {
+            const definition: CppBuildTaskDefinition = <any>_task.definition;
+            return this.getTask(definition.command, false, false, definition.args ? definition.args : [], definition);
+        }
+        return undefined;
```

Comment:
Just trying to understand what function does. Why does it return undefined if `(execution === undefined)` is false?

Reply:
As far as I know, we can implement this function to return only undefined and the behaviour of the task feature won't change for now.
The tasks defined in tasks.json file pass-through this function to be resolved (the blanks are filled-out I guess).